### DISCUSSION TITLE
查询default库表的字段时获取到重复的列名

### DISCUSSION THREAD

**Discussion by user-GitHub-user** [apache/kyuubi#4794] [Q&A] [answered]
kyuubi:1.7.0
spark:3.2.1

使用 connection.getMetaData().getColumns 方法, 除了default库外，其他库均正常无重复列

```
@Test
    public void testKyuubiJDBC() throws SQLException {
        try (Connection connection = DriverManager.getConnection("jdbc:hive2://dev1:2181,dev2:2181,dev3:2181/;"
            + "serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=kyuubi#kyuubi.engine.share.level=SERVER")) {
            ResultSet resultSet = connection.getMetaData().getColumns("spark_catalog", "default", "table",
                null);
            int length = resultSet.getMetaData().getColumnCount();
            for (int i = 1; i <= length; i++) {
                System.out.println(resultSet.getMetaData().getColumnName(i));
            }
            while (resultSet.next()) {
                for (int i = 1; i <= length; i++) {
                    System.out.print(resultSet.getString(i));
                    System.out.print("  ");
                }
                System.out.println();
            }
        }
    }
```

**Comment by cxzl25** [comment]
Using the provided version, failed to reproduce, can you provide a little more detail?

<img width="707" alt="image" src="https://user-images.githubusercontent.com/3898450/236595190-8637116d-4602-4251-9ba2-f2562951d8a5.png">

<img width="1193" alt="image" src="https://user-images.githubusercontent.com/3898450/236595204-6d63a7f3-b8c6-469d-a0d2-6d19006b5405.png">

**Reply by user-GitHub-user** [reply]
hi, i use spark for hive catalog, my database and table info:
![image](https://user-images.githubusercontent.com/18843801/236602656-a2ed34c5-ed90-4dc2-bc5c-1326a1d7e1f4.png)

and my spark catalog config(i also use iceberg):

spark.sql.catalog.spark_catalog.type = hive
spark.hive.metastore.uris=thrift://dev2:9083,thrift://dev3:9083,thrift://dev1:9083
spark.sql.catalog.spark_catalog=org.apache.iceberg.spark.SparkSessionCatalog
spark.sql.catalog.spark_catalog.warehouse=hdfs://namespace-HA-1/user/hive/warehouse

![image](https://user-images.githubusercontent.com/18843801/236602783-48b841a7-00d3-4694-b4ff-304ab43d0a2c.png)

**Reply by cxzl25** [reply]
Because iceberg's default namespace is `default`, kyuubi does not really deduplicate the namespace deduplication logic, resulting in repeated fields being returned.

This should be a bug, are you interested in submitting a PR to fix it?

https://github.com/apache/kyuubi/blob/3108c8e1a599d07e9637fb31cd09eba0d6bb0d56/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v3_0.scala#L87

https://github.com/apache/kyuubi/blob/3108c8e1a599d07e9637fb31cd09eba0d6bb0d56/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v3_0.scala#L106-L113

**Reply by cxzl25** [reply, marked as answer]
https://github.com/apache/kyuubi/issues/5105 Solved this duplicate column issue.
